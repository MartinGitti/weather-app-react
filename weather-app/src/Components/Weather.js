import React from "react"; // Import React library.
import "../App.css"; // Import React Style Sheet.

// Weather component that returns the following depending on weather results:
const Weather = props => {
  return (
    <div className="container text-light">
      <div className="Card">
        <h1 className="text-white py-3">{props.cityname}</h1> {/* User City */}
        <h5 className="py-4">
          <i className={`wi ${props.weatherIcon} display-1`} /> {/* Weather Icon */}
        </h5>

        {/* Get Celsius */}
        {props.temp_celsius ? (
          <h1 className="py-2">{props.temp_celsius}&deg;</h1>
        ) : null}

        {/* Pass in function below to show min & max temp */}
        {maxminTemp(props.temp_min, props.temp_max)}
      </div>
    </div>
  );
};

export default Weather;

// Function that displays min and max temp on users city:
function maxminTemp(min, max) {
  if (max && min) {
    return (
      <h3>
        <span className="px-4">{min}&deg;</span>
        <span className="px-4">{max}&deg;</span>
      </h3>
    );
  }
}