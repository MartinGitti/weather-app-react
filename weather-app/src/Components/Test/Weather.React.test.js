import React from 'react';
import Weather from '../Weather';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer 
    .create(<Weather/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});