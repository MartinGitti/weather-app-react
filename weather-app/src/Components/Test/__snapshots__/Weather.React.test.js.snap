// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container text-light"
>
  <div
    className="Card"
  >
    <h1
      className="text-white py-3"
    />
     
    <h5
      className="py-4"
    >
      <i
        className="wi undefined display-1"
      />
       
    </h5>
  </div>
</div>
`;
